compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -667,7 +678,19 @@ public void visit(BLangJSONArrayLiteral arrayLiteral) {
         arraySizeLiteral.value = (long) argExprs.size();
         arraySizeLiteral.type = symTable.intType;
         genNode(arraySizeLiteral, this.env);
-        emit(InstructionCodes.JSONNEWARRAY, arrayLiteral.regIndex, arraySizeLiteral.regIndex);
+
+        BLangLiteral sealedSizeLiteral = new BLangLiteral();
+        sealedSizeLiteral.pos = arrayLiteral.pos;
+        sealedSizeLiteral.value = -1L;
+        if (arrayLiteral.type.tag == TypeTags.ARRAY &&
+                ((BArrayType) arrayLiteral.type).state != BArrayState.UNSEALED) {
+            sealedSizeLiteral.value = (long) ((BArrayType) arrayLiteral.type).size;
+        }
+        sealedSizeLiteral.type = symTable.intType;
+        genNode(sealedSizeLiteral, this.env);
```

Comment:
Shall we move this common logic (writing the integer literal part) to a helper method? there are three places the same logic is used, so we can replace them..

Reply:
Addressed in 2e06894